basket/base/rq.py

```diff
@@ -0,0 +1,230 @@
+import os
+import random
+import re
+import traceback
+from time import time
+
+from django.conf import settings
+
+import redis
+import requests
+import sentry_sdk
+from django_statsd.clients import statsd
+from rq import Callback, Retry, SimpleWorker
+from rq.queue import Queue
+from rq.serializers import JSONSerializer
+from silverpop.api import SilverpopResponseException
+
+from basket.base.exceptions import RetryTask
+from basket.news.backends.common import NewsletterException
+
+# don't propagate and don't retry if these are the error messages
+IGNORE_ERROR_MSGS = [
+    "INVALID_EMAIL_ADDRESS",
+    "InvalidEmailAddress",
+    "An invalid phone number was provided",
+    "No valid subscribers were provided",
+    "There are no valid subscribers",
+    "email address is suppressed",
+    "invalid email address",
+]
```

Comment:
Some of these are probably legacy too. Because we own the CTMS client fully we could probably create and throw specific exceptions and catch and bubble the ones we want that way. Though that would be harder for Acoustic.

Reply:
That seems worthwhile. I could create an issue for it that we can add post-merge.

Reply:
https://github.com/mozmeao/basket/issues/1081